bluepysnap/_plotting.py

```diff
@@ -0,0 +1,363 @@
+            props["pop_separators"].append(props["node_id_offset"])
+        elif pd.api.types.is_categorical_dtype(spikes.nodes.property_dtypes[y_axis]):
+            props["categorical_values"].update(spikes.nodes.property_values(y_axis))
+        else:
+            props["ymin"] = min(props["ymin"], spikes.nodes.get(properties=y_axis).min())
+            props["ymax"] = max(props["ymax"], spikes.nodes.get(properties=y_axis).max())
+
+    report = filtered_report.report
+
+    dtype = spike_report[population_names[0]].nodes.property_dtypes[y_axis] if y_axis else None
+    if dtype and pd.api.types.is_categorical_dtype(dtype):
+        # this is to prevent the problems when concatenating categoricals with unknown categories
+        dtype = str
+    data = pd.Series(index=report.index, dtype=dtype)
+    for population in population_names:
+        spikes = spike_report[population]
+        mask = report["population"] == population
+        if y_axis is None:
+            data.loc[mask] = report.loc[mask, "ids"] + props["node_id_offset"]
+        else:
+            ids = report.loc[mask, "ids"].to_numpy()
+            try:
+                ys = spikes.nodes.get(properties=y_axis)
+            except BluepySnapError:
+                continue
+            # astype is used to avoid problems with the categorical
+            data[mask] = ys[ids].astype(dtype).to_numpy()
+        _update_raster_properties()
+
+    data = data[data.notna()]
+    if ax is None:
+        ax = plt.gca()
+        ax.xaxis.grid()
+        ax.set_xlabel("Time [ms]")
+        ax.tick_params(axis='y', which='both', length=0)
+        ax.set_xlim(spike_report.time_start, spike_report.time_stop)
+        if y_axis is None:
+            ax.set_ylim(0, props["node_id_offset"])
+            ax.set_ylabel("nodes")
+        else:
+            if np.issubdtype(type(data.iloc[0]), np.number):
+                # automatically expended by plt if ymin == ymax
+                ax.set_ylim(props["ymin"], props["ymax"])
+            else:
+                labels = sorted(list(props["categorical_values"]))
+                ax.set_yticks(np.arange(len(labels)))
+                ax.set_yticklabels(labels)
+                if len(labels) > 1:
+                    ax.set_ylim(-0.5, len(labels) - 0.5)
+            ax.set_ylabel("{}".format(y_axis))
+
+    ax.scatter(data.index.to_numpy(), data.to_numpy(), s=10, marker='|')
+    if len(props["pop_separators"]) > 1:
+        for separator in props["pop_separators"]:
+            ax.axhline(y=separator, color='black', lw=2)
+    return ax
+
+
+def spikes_isi(filtered_report, use_frequency=False, binsize=None, ax=None):  # pragma: no cover
+    # pylint: disable=too-many-locals
+    """Interspike interval histogram.
+
+    This plots show the binned time/frequency interval between to spikes for neurons.
+
+    Args:
+        use_frequency(bool): use inverse interspike interval times (Hz)
+        binsize(float): bin size (milliseconds or Hz)
```

Comment:
What does the default of `None` mean? Include `None` in the type.

It doesn't seem we enforce `binsize` as an `int` when using time? This is inconsistent with `time_binsize` above.

Reply:
Indeed. I changed it. 
It is now consistent with the one above.